ArduCopter/mode_zigzag.cpp

```diff
@@ -41,6 +41,15 @@ bool ModeZigZag::init(bool ignore_checks)
     return true;
 }
 
+// perform cleanup required when leaving zigzag mode
+void ModeZigZag::exit()
+{
+    // The pump will stop if the flight mode is changed from ZigZag to other
+    if (g2.zigzag_auto_pump_enabled) {
```

Comment:
This needs pragma protection against SPRAYER_ENABLED being false, doesn't it?

Reply:
Fixed it, thank you.